Framework/src/InfrastructureSpecReader.cxx

```diff
@@ -107,9 +107,18 @@ TaskSpec InfrastructureSpecReader::readSpecEntry<TaskSpec>(const std::string& ta
   ts.maxNumberCycles = taskTree.get<int>("maxNumberCycles", ts.maxNumberCycles);
   ts.resetAfterCycles = taskTree.get<size_t>("resetAfterCycles", ts.resetAfterCycles);
   ts.saveObjectsToFile = taskTree.get<std::string>("saveObjectsToFile", ts.saveObjectsToFile);
+  if (taskTree.count("extendedTaskParameters") > 0) {
+    for (const auto& [runtype, subTreeRunType] : taskTree.get_child("extendedTaskParameters")) {
+      for (const auto& [beamtype, subTreeBeamType] : subTreeRunType) {
+        for (const auto& [key, value] : subTreeBeamType) {
+          ts.customParameters.set(key, value.get_value<std::string>(), runtype, beamtype);
+        }
+      }
+    }
+  }
   if (taskTree.count("taskParameters") > 0) {
```

Comment:
maybe we should complain if users use both at the same time?

Reply:
yes, I have put a warning and I actually only use `extendedTaskParameters` in such case.